xdsl/irdl/declarative_assembly_format.py

```diff
@@ -306,39 +306,22 @@ class TypeableDirective(Directive, ABC):
     """
 
     @abstractmethod
-    def set_type(self, type: Attribute, state: ParsingState) -> None: ...
-
-    @abstractmethod
-    def get_type(self, op: IRDLOperation) -> Attribute: ...
-
-
-class VariadicTypeableDirective(AnchorableDirective, ABC):
-    """
-    Directives which can set or get multiple types.
-    """
-
-    @abstractmethod
-    def set_types(self, types: Sequence[Attribute], state: ParsingState) -> None: ...
+    def parse_single_type(self, parser: Parser, state: ParsingState) -> None: ...
```

Comment:
I'm not sure on the names `parse_single_type` and `parse_many_types` but couldn't think of anything better. I didn't want to go with `parse_type` and `parse_types` as the signatures are quite similar and this could cause confusion.

Reply:
They're not fantastically beautiful but they are quite clear, which is the most important thing. One thing to consider would be `parse_type`/`parse_type_list`

Reply:
I don't think I feel strongly either way. Happy to change it if you'd like me to.